pkg/pillar/cmd/nim/nim.go

```diff
@@ -923,10 +923,10 @@ func tryDeviceConnectivityToCloud(ctx *devicenetwork.DeviceNetworkContext) bool
 			log.Warnf("tryDeviceConnectivityToCloud: remoteTemporaryFailure: %s", err)
 		} else {
 			log.Infof("tryDeviceConnectivityToCloud: Device cloud connectivity test restart timer due to %s", err)
+			ctx.CloudConnectivityWorks = false
```

Comment:
I would make a lot more sense to be if you switched the polarity and name i.e., a CloudConnectivityFails = true
In the rtf case we can then comment that it seems like the cloud is there but not functioning correctly hence we do not mark it as Fails.

Reply:
If not, then at least add the comment in the block on L923.

Reply:
added comment